server/app/filters/SessionIdFilter.java

```diff
@@ -0,0 +1,47 @@
+package filters;
+
+import static play.mvc.Results.redirect;
+
+import akka.stream.Materializer;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import play.mvc.Filter;
+import play.mvc.Http;
+import play.mvc.Result;
+
+/** Filter that stores a random session ID in the session, if not already present. */
```

Comment:
**suggestion (blocking):** please provide a little more info on the intention of the filter. E.g. "Filter that ensures all sessions have have a unique ID."